server_apis/python/example/auto_blacklist_from_watchlist.py

```diff
@@ -0,0 +1,162 @@
+import sys
+import pika
+import pprint
+import random
+import optparse
+import json
+import requests
+
+def blacklist_binary(md5):
+    """
+    Performs a POST to the Carbon Black Server API for blacklisting an MD5 hash
+    """
+    print "blacklisting md5:%s" % (md5)
+
+    global cbtoken
+    global cbserver
+
+    headers = {'X-AUTH-TOKEN': cbtoken}
+
+    data = {"md5hash": md5, 
+        "text":"Auto-Blacklist From Watchlist", 
+        "last_ban_time":0, 
+        "ban_count":0,
+        "last_ban_host":0,
+        "enabled":True }
+
+    r = requests.post("https://%s/api/v1/banning/blacklist" % (cbserver), 
+        headers = headers, 
+        data = json.dumps(data),
+        verify = False)
+
+    if r.status_code == 409:
+        print "This md5 hash is already blacklisted"
+    elif r.status_code == 200:
+        print "Carbon Black Server API Success"
+    else:
+        print "CarbonBlack Server API returned an error: %d" % (r.status_code)
+        print "Be sure to check the Carbon Black API token"
+
+def on_message(channel, method_frame, header_frame, body):
+    """
+    Callback function which filters out the feeds we care about.
+    """
+
+    try:
+        if "application/json" == header_frame.content_type:
+
+            if method_frame.routing_key == 'watchlist.hit.binary':
+                print "watchlist.hit.binary consume"
+                
+                parsed_json = json.loads(body)
+                print parsed_json
+                lst = parsed_json['docs']
+                for item in lst:
+                    blacklist_binary(item['md5'])
+            
+            elif method_frame.routing_key == 'watchlist.hit.process':
```

Comment:
Note that this will ban every single new binary since one of the built in watchlists hits on every new binary